packages/core/src/browser/storage-service.ts

```diff
@@ -57,7 +61,17 @@ export class LocalStorageService implements StorageService {
 
     setData<T>(key: string, data?: T): Promise<void> {
         if (data !== undefined) {
-            this.storage[this.prefix(key)] = JSON.stringify(data);
+            try {
+                if (!this.isLocalStorageFull()) {
+                    this.storage[this.prefix(key)] = JSON.stringify(data);
+                }
+            } catch (e) {
+                if (this.messageService) {
+                    this.fullstorage = true;
+                    const currentSize = this.verifyLocalStorage();
+                    this.messageService.warn(`Web storage quota: \n  ${e} \n current size: ${currentSize} bytes`);
```

Comment:
I don't think this message helps the user, as it doesn't explain what it is about.
Also it might be that there are already many old buckets under the same URI. We might better ask the user whether we should clean 'old' data.

Reply:
This issue was just to detect when there is no more local storage availble to the end user. Before the issue fix by @akosyakov ( commit a5f78521a7b9ec2a190111877db70932f3f0c829 [markers] don't store by default ) offerred on July 10, it was really easy to reach the local storage quota when compiling Java project.
Should we put something like: "Proposed message: You web local storage is full < size>, you need to remove some of the data stored to allow new data to be saved. The new data could be for example you current working layout." until we fixed issue #2272 ?
The issue #2272 is to find a solution to let the end user decide what to do . Some solutions could be : 
- clear his web cache (not necessary the best solution)
- propose to the user a list of data already stored then let him decide what should be removed.
- clean up automatically the workspace that does not exist anymore